pilot/pkg/networking/core/v1alpha3/gateway_test.go

```diff
@@ -2263,3 +2263,115 @@ func TestBuildNameToServiceMapForHttpRoutes(t *testing.T) {
 		t.Errorf("The value of hostname %s mapping must be exist and it should be nil.", bazHostName)
 	}
 }
+
+func TestBuildGatewayListenersFilters(t *testing.T) {
+	cases := []struct {
+		name                   string
+		node                   *pilot_model.Proxy
+		gateways               []config.Config
+		virtualServices        []config.Config
+		expectedHTTPFilters    []string
+		expectedNetworkFilters []string
+	}{
+		{
+			"http",
+			&pilot_model.Proxy{Type: pilot_model.Router},
+			[]config.Config{
+				{
+					Meta: config.Meta{Name: uuid.NewString(), Namespace: uuid.NewString(), GroupVersionKind: gvk.Gateway},
+					Spec: &networking.Gateway{
+						Servers: []*networking.Server{
+							{
+								Port: &networking.Port{Name: "http", Number: 80, Protocol: "HTTP"},
+							},
+						},
+					},
+				},
+			},
+			nil,
+			[]string{"istio.metadata_exchange", "istio.alpn", "envoy.filters.http.fault", "envoy.filters.http.cors", "envoy.filters.http.router"},
+			[]string{"envoy.filters.network.http_connection_manager"},
+		},
+		{
+			"gateway with passthrough",
+			&pilot_model.Proxy{Type: pilot_model.Router},
+			[]config.Config{
+				{
+					Meta: config.Meta{Name: "passthrough-gateway", Namespace: "testns", GroupVersionKind: gvk.Gateway},
+					Spec: &networking.Gateway{
+						Servers: []*networking.Server{
+							{
+								Port:  &networking.Port{Name: "tcp", Number: 9443, Protocol: "TLS"},
+								Hosts: []string{"barone.example.com"},
+								Tls:   &networking.ServerTLSSettings{CredentialName: "test", Mode: networking.ServerTLSSettings_PASSTHROUGH},
```

Comment:
why passthrough? I thought we had decided to do it for istio_mutual only?

Reply:
yeah, that we discussed for setting alpn in another PR. And same rationale applies here as well while adding MX filter and should be added for ISTIO_MUTUAL only.

That led me to think why we are adding httpMx filter ALWAYS, https://github.com/istio/istio/blob/a48b0e7ded79bb16aceaf78cdd9bdd70cdb51934/pilot/pkg/networking/core/v1alpha3/listener.go#L1446 instead of adding only for ISTIO_MUTUAL case?